From f138908c22c2dcef0186a77da709e0683e8f5ec8 Mon Sep 17 00:00:00 2001
From: Yonggang Luo <luoyonggang@gmail.com>
Date: Fri, 3 Dec 2021 03:14:45 +0800
Subject: [PATCH] win32: The opengl headers should install to mesa/GL subdir

The subdir `mesa/GL` used to avoid conflict between Windows GL headers and mesa GL headers.
The GL headers resident in Windows SDK mingw dir `x86_64-w64-mingw32\include\GL` are:
```
gl.h
glaux.h
glcorearb.h
glext.h
glu.h
glxext.h
wgl.h
wglext.h
```

Signed-off-by: Yonggang Luo <luoyonggang@gmail.com>
---
 include/meson.build | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/include/meson.build b/include/meson.build
index d1d5787cf1d8..f4686cd3eb09 100644
--- a/include/meson.build
+++ b/include/meson.build
@@ -62,11 +62,15 @@ if not with_glvnd
   endif
 
   if with_opengl
+    opengl_subdir = 'GL'
+    if host_machine.system() == 'windows'
+      opengl_subdir = 'mesa/GL'
+    endif
     install_headers(
       'GL/gl.h',
       'GL/glcorearb.h',
       'GL/glext.h',
-      subdir : 'GL',
+      subdir : opengl_subdir,
     )
   endif
 
-- 
GitLab

